src/Markup/Avalonia.Markup/Markup/Parsers/SelectorParser.cs

```diff
@@ -138,16 +138,16 @@ public SelectorParser(Func<string, string, Type> typeResolver)
 
                         break;
                     case SelectorGrammar.ChildSyntax child:
-                        result = result.Child();
+                        result = result!.Child();
```

Comment:
Should probably throw an error in these cases here, but given that this is pretty deep in the internal machinery, and shouldn't happen anyway, I just decided to add `!` for now.